### ISSUE TITLE
Fine tuning with a new loss

### ISSUE THREAD

**Issue by yanaiela** [allenai/allennlp#2728]
The current fine-tune command seems very strict and does not allow too much variance on what we can do with it.
I have a use case where I'd like to fine tune my model, but with a slightly different loss function. As far as I understand, the current command won't allow it as it loads the previous model as is.
It would be nice to be able to override the previous model.
For example, by adding an override option in the config file.

Currently, I guess this can be solved by training a new model and loading the weights in the constructor, as suggested here: https://github.com/allenai/allennlp/issues/1164, but I think it would be nice to get it from the framework.

**Comment by matt-gardner** [comment]
There is an `--overrides` commandline argument for the `fine-tune` command - does this do what you want?

**Comment by yanaiela** [comment]
Based on this: https://github.com/allenai/allennlp/issues/2562, I thought that it will be ignored. Is providing the `overrides` argument will actually happen after the model was loaded already? but then wouldn't I also need to take care of the pre-trained weights loading

**Comment by matt-gardner** [comment]
Hmm, I think we should be passing `overrides` to the call to `load_archive` here, like we do for the `Params.from_file` call just below it: https://github.com/allenai/allennlp/blob/deae1db203efa9400b1d8b479b4faca3876a907a/allennlp/commands/fine_tune.py#L140-L141

Or perhaps have a separate `--model-overrides` flag, or something, to fix this.  PR welcome!

**Comment by DeNeutoy** [comment]
This should be a script, it's hard to support arbitrary different ways of fine-tuning.
